hnn_core/gui/_viz_manager.py

```diff
@@ -795,16 +847,56 @@ def compose(self):
         ])
         return config_panel, fig_output_container
 
+    def layout_template_change(self, template_type):
+        # check if plot set type requires loaded sim-data
+        if check_template_type_is_data_dependant(template_type.new):
+            # Add only simualated data
+            sim_names = [simulations for simulations, sim_name
+                         in self.data["simulations"].items()
+                         if sim_name['net'] is not None]
+
+            if len(sim_names) == 0:
+                sim_names = [" "]
+
+            self.datasets_dropdown.options = sim_names
+            self.datasets_dropdown.value = sim_names[0]
+            # show list of simulated to gui dropdown
+            self.datasets_dropdown.layout.visibility = "visible"
+        else:
+            # hide sim-data dropdown
+            self.datasets_dropdown.layout.visibility = "hidden"
+
     @unlink_relink(attribute='figs_config_tab_link')
     def add_figure(self, b=None):
         """Add a figure and corresponding config tabs to the dashboard.
         """
+        template_name = self.widgets['templates_dropdown'].value
+        is_data_template = check_template_type_is_data_dependant(template_name)
+        # if it's a data dependent layout use data_templates dictionary
+        template_type = (fig_templates[template_name]
+                         if not is_data_template
+                         else data_templates[template_name])
+
+        # set up plot fig according to template params
         _add_figure(None,
                     self.widgets,
                     self.data,
+                    template_type,
                     scale=0.97,
                     dpi=self.viz_layout['dpi'])
 
+        # check if plotting a data dependent template fig
+        num_data_sims = len(self.data["simulations"])
```

Comment:
I noticed that if `is_data_template` = True and `num_data_sims` == 0, a blank plot still is added and you can't get rid of it because the figure controls don't populate. 

Perhaps we should wrap `_add_figure` in a conditional so nothing is generated if certain conditions are not met?

Reply:
I refactored the add_figure code. It validates the existence of data to create figures.